osdconfig/api.go

```diff
@@ -185,39 +189,74 @@ func (manager *configManager) SetNodeConf(config *NodeConfig) error {
 	defer manager.Unlock()
 
 	if config == nil {
-		return InputErr
+		return ErrorInput
 	}
 
 	if len(config.NodeId) == 0 {
-		return InputErr
+		return ErrorInput
 	}
 
-	// push into kvdb
+	// push node data into kvdb
 	if _, err := manager.cc.Put(getNodeKeyFromNodeID(config.NodeId), config, 0); err != nil {
 		return err
 	}
+
+	// update node index
+	list, err := manager.GetNodeList()
+	if err != nil {
+		list = make([]string, 0, 0)
+	}
+	list = append(list, config.NodeId)
+	list = dedupe(list)
```

Comment:
Do not use dedup or this type of list. Instead create a new class called Set, which additions automatically make sure there is only one. This is how it works in python. Many years ago I created one which you can grab:

https://github.com/heketi/utils/blob/master/stringset.go

Reply:
done. since I am storing this data in kvdb as slice of string, it is easier to work with that data type. I did, however, find a way to not use dedupe.